### PR TITLE
feat: enhance add_nodes method to support nested children structure

### PR DESCRIPTION
**Pull request by UmbraCi**
## Overview

This PR enhances the `add_nodes` method to support nested children structures from all major jsMind data formats (node_tree, node_array, and freemind), while maintaining complete backward compatibility.

### ✅ Backward Compatibility
- All existing code using the legacy format continues to work unchanged
- All existing tests pass without modification
- API signature remains the same

### 🌳 Nested Structure Support
- Supports `children` arrays with recursive processing
- Handles deep nesting (unlimited levels)
- Processes all child nodes in a single operation

### 📊 Multiple Format Support
- **node_tree format**: Native nested object structure
- **node_array format**: Custom attributes with children
- **freemind format**: XML-style attributes preserved
- **Legacy format**: Original format continues to work

### ⚡ Performance Optimized
- Single UI refresh after all nodes are added
- Batch processing for better performance
- Efficient recursive processing

### Legacy Format (Unchanged)
```javascript
// This continues to work exactly as before
jm.add_nodes('parent_id', [
    {
        node_id: 'child1',
        topic: 'Child Node 1',
        data: { color: 'red' },
        direction: 'left'
    }
]);
```

### New Nested Format
```javascript
// Support for nested children
jm.add_nodes('root', [
    {
        id: 'category',
        topic: 'Main Category',
        direction: 'right',
        children: [
            {
                id: 'subcategory1',
                topic: 'Subcategory 1',
                children: [
                    { id: 'item1', topic: 'Item 1' },
                    { id: 'item2', topic: 'Item 2' }
                ]
            }
        ]
    }
]);
```

### New Methods Added
- `_add_nodes_recursive(parent_node, node_data)`: Recursively process nodes and children
- `_extract_node_tree_subnode(node_data)`: Extract children from various formats

### Format Detection
The method automatically detects the format:
1. **Nested Format**: If `node_data.children` exists and is an array
2. **Legacy Format**: Otherwise, uses original legacy processing
3. **Hybrid Support**: Can process both formats in the same call

### For Developers
- **Easier Data Import**: Directly use data from node_tree, node_array, or freemind formats
- **Reduced Code**: No need to manually flatten nested structures
- **Better Performance**: Single operation with batch UI refresh
- **Future-Proof**: Ready for complex data structures

### For Applications
- **Rich Data Support**: Preserve custom attributes and metadata
- **Format Flexibility**: Support multiple input formats seamlessly
- **Migration Path**: Easy upgrade from legacy to nested formats
- **Interoperability**: Better integration with other mind mapping tools

## Testing

The enhancement includes comprehensive tests covering:
- All legacy format scenarios (100% backward compatibility)
- Nested children processing
- Mixed format handling
- Error cases and edge conditions
- Performance with large datasets
- Custom attribute preservation

## Files Changed

- `src/jsmind.js`: Enhanced add_nodes method with nested support
- `tests/unit/jsmind.test.js`: Comprehensive test coverage
- `docs/add_nodes_enhancement.md`: Detailed documentation

### No Changes Required
Existing code continues to work without any modifications.

### Optional Enhancements
To take advantage of new features, developers can now directly use nested data structures instead of manually flattening them.

This enhancement makes jsMind's `add_nodes` method significantly more powerful while maintaining complete backward compatibility.

### CODE REVIEW

**Comment by UmbraCi** [comment]
> 加了些评论，麻烦看看。谢谢！

重构了一下这个方法，因为周末在家是Windows电脑，所以需要加一下cross依赖用来运行script。
另外想问下，开发这个库的时候，我看example中的文件引用的script和css都是cdn的，如果我想一边开发一边调试，你一般是怎么做的呢？ 新建一个demo应用，然后包pnpm link? @hizzgdev

**Comment by hizzgdev** [comment]
> > 加了些评论，麻烦看看。谢谢！
> 
> 重构了一下这个方法，因为周末在家是Windows电脑，所以需要加一下cross依赖用来运行script。 另外想问下，开发这个库的时候，我看example中的文件引用的script和css都是cdn的，如果我想一边开发一边调试，你一般是怎么做的呢？ 新建一个demo应用，然后包pnpm link? @hizzgdev

你看 example/demo.html 这里面引用的是 es6 目录里的文件，你在本地用 `npm run build` 就可以了。它生成的有 map 文件也不影响调试。

File: src/jsmind.js [hunk 1]
```diff
@@ -504,33 +505,169 @@ export default class jsMind {
...
+            }
+
+            // filter null values
+            all_created_nodes = all_created_nodes.filter(node => node !== null);
+
+            const expected_count = this._count_expected_nodes(nodes_data);
+            const actual_count = all_created_nodes.length;
+
+            // Validate node creation success and handle cleanup if needed
+            if (actual_count > 0) {
```
Comment (hizzgdev) [inline src/jsmind.js:523]: `actual_count > 0` 这个条件我觉得没有必要，因为 `expected_count` 一定会 `>0` 因为前面已经判断了 `nodes_data.length === 0`

File: src/jsmind.js [hunk 2]
```diff
@@ -504,33 +505,169 @@ export default class jsMind {
...
+                        evt: 'add_nodes',
+                        data: [the_parent_node.id, nodes_data],
+                        nodes: all_created_nodes.map(node => node.id),
+                    });
+                } else {
+                    // Some nodes failed, clean up partially created nodes
+                    logger.warn(
+                        `Expected ${expected_count} nodes, but only created ${actual_count}. Cleaning up...`
+                    );
+                    this._cleanup_partial_nodes(all_created_nodes);
```
Comment (hizzgdev) [inline src/jsmind.js:537]: 我们需要考虑一下这个方法需要在什么时候调用，是判断数量不一致的时候，还是在 catch(exception) 的时候，我觉得它似乎不需要在两处都调用。

**Review by hizzgdev — COMMENTED** [review]
麻烦再看看，谢谢！

File: src/jsmind.js [hunk 3]
```diff
@@ -504,33 +505,169 @@ export default class jsMind {
...
+    _count_expected_nodes(nodes_data) {
+        return nodes_data.reduce((count, node_data) => {
+            count++; // Count current node
+            const children = this._extract_node_tree_subnode(node_data);
+            if (children && children.length > 0) {
+                count += this._count_expected_nodes(children);
+            }
+            return count;
+        }, 0);
+    }
```
[the remark this answers is not in the document]
Comment (hizzgdev) [reply src/jsmind.js:614]: 你注意到这个 comment 了吗？

File: src/jsmind.js (hunk 1 above)
Comment (hizzgdev) [reply src/jsmind.js:525]: 因为后面有 `if (actual_count === expected_count)` 而且其中的 expected_count 一定会 > 0 ，所以我们并不需要单独判断 actual_count > 0, 因为如果 acount_count <= 0 的话，那一定不会满足后面的 `actual_count === expected_count`

File: tests/unit/jsmind.test.js [hunk 4]
```diff
@@ -558,362 +558,161 @@ describe('add_nodes', () => {
...
+    return {
+        id: id,
+        topic: topic,
+        children: [],
+        isroot: false,
+        parent: null,
+        direction: 1,
+        expanded: true,
+        data: {},
+        _data: {
```
Comment (hizzgdev) [inline tests/unit/jsmind.test.js:706]: 这个 `_data` 也需要吗？

File: package.json [hunk 5]
```diff
@@ -37,9 +37,9 @@
         "server": "http-server",
         "build": "rollup -c .config/rollup.config.js",
         "build-types": "npx -p typescript tsc -p types/tsconfig.declaration.json",
-        "test": "NODE_OPTIONS=--experimental-vm-modules jest tests/unit",
-        "test-es6": "NODE_OPTIONS=--experimental-vm-modules jest tests/unit",
-        "test-types": "NODE_OPTIONS=--experimental-vm-modules jest tests/types",
+        "test": "cross-env NODE_OPTIONS=--experimental-vm-modules jest tests/unit",
+        "test-es6": "cross-env NODE_OPTIONS=--experimental-vm-modules jest tests/unit",
+        "test-types": "cross-env NODE_OPTIONS=--experimental-vm-modules jest tests/types",
```
Comment (hizzgdev) [inline package.json:42]: 这个文件还原回来吧。

**Review by hizzgdev — COMMENTED** [review]
再次感谢 @UmbraCi 的支持。
这次的更新挺多，或许还有一些 bug，这个 PR 合并之后，我们需要把这个方法加到文档里，到时候在文档中把它标记为 [实验性], 以提醒用户小心使用。

**Comment by UmbraCi** [comment]
麻烦再cr下 @hizzgdev

**Review by hizzgdev — APPROVED** [review]
